flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliStrings.java

```diff
@@ -36,91 +43,149 @@ private CliStrings() {
+
+        public SQLCliCommandsDescriptions() {
+            commandsDescriptions = new ArrayList<>();
+        }
+
+        public SQLCliCommandsDescriptions commandDescription(
+                Tuple2<String, String> commandDescription) {
+            Preconditions.checkState(
+                    checkCommandDescription(commandDescription),
+                    "All contents of command description must not be empty.");
+            this.updateMaxCommandLength(commandDescription.f0.length());
+            this.commandsDescriptions.add(commandDescription);
+            return this;
+        }
+
+        private void updateMaxCommandLength(int newLength) {
+            Preconditions.checkState(newLength > 0);
+            if (this.commandMaxLength < newLength) {
+                this.commandMaxLength = newLength;
+            }
+        }
+
+        private boolean checkCommandDescription(Tuple2<String, String> commandDescription) {
+            return Objects.nonNull(commandDescription)
+                    && StringUtils.isNotBlank(commandDescription.f0)
+                    && StringUtils.isNotBlank(commandDescription.f1);
+        }
+
+        public void clearCommandsDescriptions() {
+            this.commandsDescriptions.clear();
+            this.resetMaxCommandLength();
+        }
+
+        private void resetMaxCommandLength() {
+            this.commandMaxLength = -1;
+        }
+
+        public int getCommandMaxLength() {
+            return this.commandMaxLength;
+        }
+
+        public AttributedString formatCommand(
+                String cmdDescDelimiter,
+                String cmdFormat,
+                String descFormat,
+                boolean clearCmdsDescs) {
+            Preconditions.checkNotNull(cmdDescDelimiter, "delimiter must not be null.");
+            Preconditions.checkState(
+                    StringUtils.isNotBlank(cmdFormat), "command format must not be null.");
+            Preconditions.checkState(
+                    StringUtils.isNotBlank(descFormat),
+                    "command description format must not be null.");
+            AttributedStringBuilder attributedStringBuilder = new AttributedStringBuilder();
+            if (!this.commandsDescriptions.isEmpty()) {
+                this.commandsDescriptions.forEach(
+                        cmdDescTuple2 -> {
+                            attributedStringBuilder
+                                    .style(AttributedStyle.DEFAULT.bold())
+                                    .append(String.format(cmdFormat, cmdDescTuple2.f0))
+                                    .append(CMD_DESC_DELIMITER)
+                                    .style(AttributedStyle.DEFAULT)
+                                    .append(String.format(descFormat, cmdDescTuple2.f1))
+                                    .append('\n');
+                        });
+            }
+            if (clearCmdsDescs) {
+                this.clearCommandsDescriptions();
```

Comment:
Why we need to clear the command descriptions? We don't allow the `formatCommand` to be called multiple times?